src/trace_processor/importers/proto/winscope/winscope_transform_tracker.cc

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "src/trace_processor/importers/proto/winscope/winscope_transform_tracker.h"
+
+#include <algorithm>
+#include "src/trace_processor/storage/trace_storage.h"
+
+namespace perfetto::trace_processor::winscope {
+
+tables::WinscopeTransformTable::Id* WinscopeTransformTracker::GetOrInsertRow(
+    geometry::TransformMatrix& matrix) {
+  auto existing_matrix = std::find_if(
```

Comment:
Do you really want to do a linear scan every time you call this function? Why is this not using a base::FlatHashMap?